src/astro/utils/dependencies.py

```diff
@@ -1,86 +1,97 @@
 from astro import constants
 
 
-class MissingPackage:
-    def __init__(self, module_name, related_extras):
-        self.module_name = module_name
-        self.related_extras = related_extras
+class MissingPackage(type):
+    """
+    Base class used to handle missing packages.
+    Raises an exception if the user attempts to access a class attribute.
+    """
 
-    def __getattr__(self, item):
+    package_name: str = ""
+    related_extras: str = ""
+
+    def __getattr__(cls, key: str) -> None:
         raise RuntimeError(
-            f"Error loading the module {self.module_name},"
+            f"Error loading the package {cls.package_name},"
             f" please make sure all the dependencies are installed."
-            f" try - pip install {constants.PYPI_PROJECT_NAME}[{self.related_extras}]"
+            f" try - pip install {constants.PYPI_PROJECT_NAME}[{cls.related_extras}]"
         )
 
 
-try:
-    from airflow.providers.google.cloud.hooks.bigquery import BigQueryHook
-except ModuleNotFoundError:
-    BigQueryHook = MissingPackage(
-        "airflow.providers.google.cloud.hooks.bigquery", "google"
-    )
+class GoogleMissingPackage(metaclass=MissingPackage):
+    """Class used to represent missing dependencies related to Google services."""
 
-try:
-    from airflow.providers.postgres.hooks.postgres import PostgresHook
-except ModuleNotFoundError:
-    PostgresHook = MissingPackage(
-        "airflow.providers.postgres.hooks.postgres", "postgres"
-    )
+    package_name = "apache-airflow-providers-google"
+    related_extras = "google"
 
-try:
-    from airflow.providers.snowflake.hooks.snowflake import SnowflakeHook
-except ModuleNotFoundError:
-    SnowflakeHook = MissingPackage(
-        "airflow.providers.snowflake.hooks.snowflake", "snowflake"
-    )
 
-try:
-    from snowflake.connector import pandas_tools
-except ModuleNotFoundError:
-    pandas_tools = MissingPackage("snowflake-connector-python[pandas]", "postgres")
+class AmazonMissingPackage(metaclass=MissingPackage):
+    """Class used to represent missing dependencies related to Amazon services."""
 
-try:
-    from boto3 import Session as BotoSession
-except ModuleNotFoundError:
-    BotoSession = MissingPackage("s3fs", "amazon")
+    package_name = "apache-airflow-providers-amazon"
+    related_extras = "amazon"
 
-try:
-    from google.cloud.storage import Client as GCSClient
-except ModuleNotFoundError:
-    GCSClient = MissingPackage("apache-airflow-providers-google", "google")
 
-try:
-    from google.cloud import bigquery
-except ModuleNotFoundError:
-    bigquery = MissingPackage("apache-airflow-providers-google", "google")
+class SnowflakeMissingPackage(metaclass=MissingPackage):
```

Comment:
How did creating subclasses help?

Reply:
It allowed the method `get_hook` from `astro.utils` to consistently return a class - as opposed to sometimes a class and sometimes an instance. This made MyPy happy